protocol-units/da/m1/light-node/src/v1/sequencer.rs

```diff
@@ -75,23 +80,79 @@ impl LightNodeV1 {
 			}
 			None => {
 				// no transactions to include
-				info!(target: "movement_timing", uid = %uid, "no_transactions_to_include");
+				debug!(target: "movement_timing", uid = %uid, "no_transactions_to_include");
 				Ok(())
 			}
 			
 		}
 
 	}
 
-	async fn tick_publish_blobs(&self, receiver : &mut Receiver<Block>) -> Result<(), anyhow::Error> {
-		
-		let half_building_time = self.memseq.building_time_ms();
-		let start = std::time::Instant::now();
-		let mut blocks = Vec::new();
+	/// Attempts to submit a batch of blobs according to the chunk size
+	async fn submit_and_resize(&self, block_blobs: &mut Vec<celestia_types::Blob>, chunk_size : usize) -> Result<usize, anyhow::Error> {
+	
+
+		// iter in chunks of 4
+		while !block_blobs.is_empty() {
+			let mut chunk: Vec<celestia_types::Blob> = Vec::with_capacity(chunk_size);
+			
+			for _ in 0..chunk_size {
+				if let Some(blob) = block_blobs.pop() {
+					chunk.push(blob);
+				} else {
+					break;
+				}
+			}
+			
+			match self.pass_through.submit_celestia_blobs(&chunk).await {
+				Ok(_) => {
+					info!(
+						target: "movement_timing",
+						batch_size = chunk.len(),
+						"submitted_blob_batch",
+					);
+				}
+				Err(e) => {
+					info!("failed to submit blocks: {:?}", e);
+					block_blobs.extend(chunk.into_iter());
+					return Ok(chunk_size - 1);
+				}
+			}
+		}
+	
 
-		// select receive or timeout
+		Ok(chunk_size)
 
+	}
+
+	/// Submits block blobs in chunks, beginning with a chunk size of 4
+	async fn submit_block_blobs(&self, block_blobs: &mut Vec<celestia_types::Blob>) -> Result<(), anyhow::Error> {
+		if block_blobs.len() > 0 {
 
+			let mut chunk_size = 4;
+			loop {
+
+				let new_chunk_size = self.submit_and_resize(block_blobs, chunk_size).await?;			
+				if new_chunk_size == chunk_size {
```

Comment:
This exit condition assumes particulars from the behavior of `submit_and_resize` method. I'd rather have the enclosing loop as `while block_blobs.len() != 0`, then you can also eliminate the top-level check.